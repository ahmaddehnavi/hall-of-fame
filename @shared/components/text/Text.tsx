import {observer} from 'mobx-react';
import * as React from 'react';
import {StyleSheet, Text as RNText, TextProps} from 'react-native';

type Props = TextProps & {
    bold?: boolean
    title?: string
}

@observer
export class Text extends React.Component<Props> {
    render() {

        return (
            <RNText
                {...this.props}
                children={this.props.title || this.props.children}
                style={[this.props.bold ? styles.textBold : styles.text, this.props.style]}
            />
        )
    }
}


const styles = StyleSheet.create({
    text: {
        // todo default normal font,
        textAlign: 'left'
    },
    textBold: {
        // todo default bold font,
        textAlign: 'left'
    },
});