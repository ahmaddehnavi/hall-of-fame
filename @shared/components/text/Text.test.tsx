import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {Text} from './Text';

describe('Text Tests', () => {

    it('should match snapshot', () => {
        let tree = renderer.create(
            <Text
            />
        );
        expect(tree).toMatchSnapshot();
    });

});
