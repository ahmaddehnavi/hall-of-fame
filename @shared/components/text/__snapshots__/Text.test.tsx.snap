// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Tests should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "textAlign": "left",
      },
      undefined,
    ]
  }
/>
`;
